### ISSUE TITLE
[Future] Support "script modules"

### ISSUE THREAD

**Issue by AlCalzone** [ioBroker/ioBroker.javascript#86]
Besides scripts in VMs, NodeJS 9+ supports modules in VMs:
https://nodejs.org/api/vm.html#vm_class_vm_module

This could be used to modularize code in the JS adapter, as some users have already asked for.

**Comment by GermanBluefox** [comment]
Future => Feature ?

**Comment by AlCalzone** [comment]
No, I mean future. Just as a reminder when NodeJS 9 (or rather 10) becomes the norm.
At the moment we can't use this, but it could be a very nice feature.

**Comment by AlCalzone** [comment]
Now that NodeJS 10 is available to the public, we could tackle this similar to how NodeJS 6+ is using VM2.

**Comment by GermanBluefox** [comment]
It is still experimental event in node 13

**Comment by Apollon77** [comment]
Still experimental even in nodejs 15

**Comment by AlCalzone** [comment]
🙈
